packages/core/src/utils/test-utils.ts

```diff
@@ -71,7 +77,11 @@ export const createRequester = ({
     const anonymousRouter: AnonymousRouter = new Router();
 
     for (const route of Array.isArray(anonymousRoutes) ? anonymousRoutes : [anonymousRoutes]) {
-      route(anonymousRouter, provider);
+      if (provider) {
+        route(anonymousRouter, provider);
+      } else {
+        (route as RouteLauncher<AnonymousRouter>)(anonymousRouter);
+      }
```

Comment:
instead of having `as` here, will it be better to have several function signatures (overloads) to make more precise function calls?

Reply:
Lint suggests combine those overloads, I'll try agian.

Reply:
<img width="954" alt="image" src="https://user-images.githubusercontent.com/5717882/155281529-c95d7983-cfab-4a82-b854-520c4dc17a71.png">

I tried but it won't work, any suggestions?

Reply:
per offline discussion, `as` is necessary(for runtime), but add function overloads  to infer types for IDE.